### ISSUE TITLE
Feature request: integrate tilesets-cli with GDAL/OGR tools

### ISSUE THREAD

**Issue by tjukanovt** [mapbox/tilesets-cli#153]
As a user I would like to be able to upload large datasets directly from different data sources without converting them to GeoJSON. It would be great if I could use a GeoPackage, shapefile, FGDB, SQLite or even PostGIS as the upload data source. GeoJSON is quite inefficient especially with large datasets. 

One option might be to integrate the tilesets-cli pipeline with [ogr2ogr](https://gdal.org/programs/ogr2ogr.html), which is an open source ETL CLI tool for converting data, supporting a huge number of vector datasets out of the box.

**Comment by mapsam** [comment]
@tjukanovt I agree it'd be nice to use ogr2ogr right out of the box! But it gets a little tricky when delivering the gdal binaries in the package, as you can see in the [README.md](https://github.com/mapbox/tilesets-cli#installing-optional-estimate-area-command) we don't ship GDAL by default because of A) the size and B) the complexity across different environments. 

In the meantime, you can still use GDAL libraries, like Fiona, to pipe directly to the `upload-source` command. For example, here's me converting the Natural Earth Data countries from Shapefile to line delimited geojson: 

```sh
fio cat ~/Downloads/ne_110m_admin_0_countries/ne_110m_admin_0_countries.shp | tilesets upload-source mapsam ne_110m_countries
{"id": "mapbox://tileset-source/mapsam/ne_110m_countries", "files": 1, "source_size": 970051, "file_size": 970051}
```

I'd suggest giving [Fiona](https://fiona.readthedocs.io/en/latest/README.html) a try. I'm sure the same command is possible with ogr2ogr directly as well.
